tests/kernel/usb_nic.pm

```diff
@@ -0,0 +1,48 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Package: usb_nic
+# Summary: Simple smoke test for testing USB NIC connected to system
+# Maintainer: LSG QE Kernel <kerneli-qa@suse.de>
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use utils;
+use Utils::Logging 'export_logs_basic';
+
+sub run {
+    my ($self) = @_;
+
+    select_serial_terminal;
+
+    zypper_call('in -t package ethtool');
+
+    my $interface = script_output 'basename $(readlink /sys/class/net/* | grep usb )';
+
+    assert_script_run "echo \"BOOTPROTO='dhcp'\" > /etc/sysconfig/network/ifcfg-$interface";
+    assert_script_run "echo \"STARTMODE='auto'\" >> /etc/sysconfig/network/ifcfg-$interface";
+
+    assert_script_run "ifup $interface";
+
+    # wait until interface is up
+    my $timeout = 20;
+    until (script_run "ip -4 addr show $interface | grep inet") {
```

Comment:
I guess you would like to replace `until` with `while` or maybe `do {} while`. `until` executes a block of code as long as the condition is false. So the `script_run` returns 0 which is in perl `false`. Hence in the VR you can see it run more times showing that he interface is already up with an IP